src/seeds.gratitude.cpp

```diff
@@ -70,38 +92,88 @@ ACTION gratitude::acknowledge (name from, name to, string memo) {
         item.receivers.push_back(to);
     });
   }
+
+  // Updates ack stats
+  auto stitr = stats.rbegin();
+  auto round_id = stitr->round_id;
+
+  auto stitr2 = stats.find(round_id);
+  auto oldacks = stitr2->num_acks;
+  stats.modify(stitr2, _self, [&](auto& item) {
+      item.num_acks = oldacks + 1;
+  });
+
 }
 
+
 ACTION gratitude::testacks() {
   require_auth(get_self());
 
   auto actr = acks.begin();
+
   while (actr != acks.end()) {
-    calc_acks(actr->donor);
-    actr = acks.erase(actr);
+    _calc_acks(actr->donor);
+    actr++;
   }
 }
 
-ACTION gratitude::newround() {
+ACTION gratitude::calcacks(uint64_t start) {
+  require_auth(get_self());
+
+  auto actr = start == 0 ? acks.begin() : acks.find(start);
+
+  _calc_acks(actr->donor);
+
+  actr++;
+
+  // Recursion call
+  if (actr != acks.end()) {
+    action next_execution(
+      permission_level{get_self(), "active"_n},
+      get_self(),
+      "calcacks"_n,
+      std::make_tuple(actr->donor.value)
+    );
+
+    transaction tx;
+    tx.actions.emplace_back(next_execution);
+    tx.delay_sec = 1;
+    tx.send(actr->donor.value, _self);
+  } else {
+    // Cleanup on last exec
+    auto actr2 = acks.begin();
```

Comment:
This will have the same problem as the initial issue, as it's walking the entire table to delete items...

This will fail if the number of acks is > 200 or 300 or 400  (one of these numbers)

Reply:
I would suggest to erase them as they're processed, if that is possible